sdk/core/Azure.Core/src/Serialization/AzureCoreExtensions.cs

```diff
@@ -62,6 +63,8 @@ public static class AzureCoreExtensions
         /// <summary>
         /// Return the content of the BinaryData as a dynamic type.  Please see https://aka.ms/azsdk/net/dynamiccontent for details.
         /// </summary>
+        [RequiresUnreferencedCode(MutableJsonDocument.SerializationRequiresUnreferencedCodeClass)]
```

Comment:
No concerns, but help me understand the context here?  Is it that we were ignoring these warnings somehow before but with the move to .NET 8 we can expose them now?

Reply:
@annelo-msft I was baselining them in ExpectedAotWarnings.txt since they weren't impacting any other AOT compatible libraries (monitor exporter, identity, etc). I'm just trying to chip away at the expected warnings that aren't bothering anyone else. I'll probably have a couple more PRs in the near future to chip away at some more of them.

It'll be a lot easier to find regressions if we don't have any baselined warnings.

The .NET 8 update is unrelated, just working on a bunch of small things today!

Reply:
By "baselining" do you mean suppressing warnings that were present in the text file?

Reply:
No there aren't any suppressions, it only tells the pipeline to ignore the warnings when they are reported.

The baselined warnings don't show up when an app has a dependency on identity, etc because they're not in those code paths. For an application trying to use AOT compilation, they'll only see warnings for things in the code paths they are using.